library/pod-security-policy/users/src.rego

```diff
@@ -46,9 +46,18 @@ accept_users("MustRunAs", provided_user) = res  {
 get_violation(field, params, container) = msg {
   rule := params.rule
   provided_value := get_field_value(field, container, input.review)
+  not is_array(provided_value)
   not accept_value(rule, provided_value, params.ranges)
   msg := sprintf("Container %v is attempting to run as disallowed group %v. Allowed %v: %v", [container.name, provided_value, field, params])
 }
+# SupplementalGroups is array value
+get_violation(field, params, container) = msg {
+  rule := params.rule
+  array_value := get_field_value(field, container, input.review)
```

Comment:
should we have is_array() here?

Reply:
The next line will fail if the value is not an array, so an explicit is_array check is not needed

Reply:
It could succeed if the `array_value` is assigned an object.

I realize that's not the currently allowed schema, but tending toward extra defensiveness seems wise here, as a misbehaving function risks requests being rejected due to Rego throwing an exception.

Reply:
Ah, that makes sense. Didn't realise that was the behaviour with objects as well. Updated now